Career Page

yakovlevlexandrthirdparty opened the pull request:
Career page and vacancy page

Taler777 committed:
Class Task added.

Taler777 committed:
checkpoint

Taler777 committed:
The simple list of tasks is added.
The task will be opened by reference in the next release.

Taler777 committed:
Empty page of the current task is added.

Taler777 committed:
Checkpoint - showCurrentTask method is added.

Taler777 committed:
Checkpoint - Task's page fixed.

Taler777 committed:
Checkpoint - Links on the task's page fixed.

yakovlevlexandrthirdparty committed:
Career Page

Career page and vacancy page

yakovlevlexandrthirdparty committed:
Career Controller

Creer Controller

baimurzin commented on src/main/resources/templates/career.ftl:
```diff
@@ -0,0 +1,35 @@
+    <div style="margin:20px;  float: right">
+        <h3>Вакансии</h3>
+        <ul>
+            <li><a href="vacancy?id=${careerModel.getVacancy(0).getVacancyId()}">${careerModel.getVacancy(0).getCompanyName()} ${careerModel.getVacancy(0).getPositionName()} ${careerModel.getVacancy(0).getSalary()} руб/мес</a></li>
```
А если у тебя будет 100 вакансий, или 500 или 347

Taler777 committed:
Added a task availability check

baimurzin commented on src/main/java/org/itlabel/cabinet/controller/StudentCareerController.java:
```diff
@@ -0,0 +1,40 @@
+@Controller
+public class StudentCareerController {
+    Vacancy[] vacancies = {
+            new Vacancy(0, "Google", "middle Java Developer", 250000,1),
```
Вместо обычного массива используй List list = new ArrayList<>()

Taler777 replied:
Соглашусь.
Массив имеет ограничения - нельзя изменить его размер.
Поэтому пользуйся коллекциями.

baimurzin commented on src/main/java/org/itlabel/cabinet/controller/StudentCareerController.java:
```diff
@@ -0,0 +1,40 @@
+    }
+
+    @RequestMapping("/vacancy")
+    public ModelAndView greeting(@RequestParam(value="id", defaultValue="2") Integer id) {
```
не надо default value 
 у тебя не всегда может быть объект с заранее заданным айди

baimurzin commented on src/main/java/org/itlabel/cabinet/model/CareerModel.java:
```diff
@@ -0,0 +1,22 @@
+ *@author Yakovlev Alexandr
+ */
+public class CareerModel{
+    private Vacancy[] vacancies;
```
Используй List, ArrayList
У Алексея есть видео лекция по работе с freemarker и spring mvc, можешь посомтреть его 8 урок, там многие моменты увидишь

baimurzin commented on src/main/resources/templates/career.ftl:
```diff
@@ -0,0 +1,35 @@
+    <div style="margin:20px; float: left">
+        <h3>Мои навыки</h3>
+        <ul>
+            <li>${careerModel.getUser().getSkills(0)}</li>
```
итерируй в цикле выводи

Taler777 committed:
Empty Task's constructor added.

Taler777 committed:
Task's status change added.

Taler777 committed:
Repeating the task list has been fixed.

somenonename commented:
После того как закончил разработку, можешь выделить все свои файлы в которые вносил изменения и сделать форматирование кода (ctrl+alt+L), так код будет понятнее и читаемость увеличится
[Java code style](https://google.github.io/styleguide/javaguide.html)

Taler777 committed:
Task status management added.

Taler777 committed:
Spring Security disabled.

Taler777 committed:
Merge branch 'dev' into student-cab-alexeys

Taler777 commented:
``` java
public class User {

    private String name;

    private String[] skills;
```
Для скилзов тоже используй коллекции.
Ты же не знаешь сколько навыков будет изначально, а по мере роста опыта навыки будут добавляться.

Taler777 commented on src/main/java/org/itlabel/cabinet/model/Vacancy.java:
```diff
@@ -0,0 +1,45 @@
+/*
+ *@author Yakovlev Alexandr
+ */
+public class Vacancy {
```
Нет сеттера на VacancyDescription
Еще нет VacancyDescription в конструкторе - его там может и не быть, но мало ли.
Можно предусмотреть создание вакансии с частично заполненными данными - тогда надо сделать несколько конструкторов.

Taler777 commented on src/main/resources/templates/level.ftl:
```diff
@@ -8,8 +8,9 @@
 <body>
 <h2 class="hello-title">Here is my level page</h2>
 <script src="/js/main.js"></script>
-<a href="profile">My profile</a>
```
А зачем ты удалил ссылку на страницу My profile ?
Верни ее назад.

Taler777 commented on src/main/resources/templates/chat.ftl:
```diff
@@ -8,9 +8,9 @@
 <h2 class="hello-title">Here is the chat</h2>
 <script src="/js/main.js"></script>
-
-<a href="profile">My profile</a>
```
А зачем ты удалил ссылку на страницу My profile ?
Верни ее назад.

Taler777 commented on src/main/resources/templates/tasks.ftl:
```diff
@@ -8,8 +8,9 @@
 <body>
 <h2 class="hello-title">Here are my tasks</h2>
 <script src="/js/main.js"></script>
-<a href="profile">My profile</a>
```
А зачем ты удалил ссылку на страницу My profile ?
Верни ее назад.

Taler777 commented on src/main/resources/templates/vacancy.ftl:
```diff
@@ -0,0 +1,23 @@
+<body>
+<h2 class="hello-title">Here is the vacancy page</h2>
+<script src="/js/main.js"></script>
+<a href="level">My level</a>
```
Здесь было бы неплохо тоже видеть ссылку на страницу My profile ?
Добавь.

Taler777 committed:
variables renamed

Taler777 committed:
variables renamed

Taler777 committed:
variables renamed

Taler777 committed:
id type changed from int to Long
number variable added

Taler777 committed:
added: the status change with the fixed list of values

Taler777 committed:
updated

Taler777 committed:
updated

Taler777 reviewed (changes requested):
Верни ссылки на My profile, добавь сеттер, используй коллекции.
(тестирую request changes)

yakovlevlexandrthirdparty committed:
Corrections

1. link My profile
2. Array to List
3. VacansyDescription sek setter
4. Default value on get rquest

yakovlevlexandrthirdparty committed:
Career.ftl

вывел в цыкле вакансии и навыки, смотрю как убрать копирование

yakovlevlexandrthirdparty committed:
Разрастание списка

вынес инициализацию демо данных за пределы метода

yakovlevlexandrthirdparty committed:
Убрал дублимрование вакансии

yakovlevlexandrthirdparty committed:
getVacancyId

Taler777 committed:
status check to null added

Taler777 committed:
status check to null and empty string added

Taler777 committed:
status check to null and empty string added

baimurzin commented on src/main/resources/templates/vacancy.ftl:
```diff
@@ -0,0 +1,24 @@
+</head>
+<body>
+<h2 class="hello-title">Here is the vacancy page</h2>
+<script src="/js/main.js"></script>
```
Не подключай js по середине body, либо в head либо перед закрытием body.
Если не знаешь почему или интересно пиши - расскажу(собственно это касается ко всем комментариям в ревью)

baimurzin commented on src/main/resources/templates/vacancy.ftl:
```diff
@@ -0,0 +1,24 @@
+<a href="tasks">My tasks</a>
+<a href="chat">Chat</a>
+<a href="career">My career</a>
+<div style="margin:20px;">
```
По хорошему лучше выносить в файл стилей css. Но сейчас можешь и так оставить если не хочешь переносить

baimurzin commented on src/main/resources/templates/career.ftl:
```diff
@@ -0,0 +1,35 @@
+</head>
+<body>
+<h2 class="hello-title">My career</h2>
+<script src="/js/main.js"></script>
```
тоже самое

baimurzin commented on src/main/resources/templates/career.ftl:
```diff
@@ -0,0 +1,35 @@
+        <h3>Вакансии</h3>
+        <ul>
+        <#list careerModel.getVacancy() as vacancy>
+            <li><a href="vacancy?id=${vacancy.getVacancyId()}">${vacancy.getCompanyName()} ${vacancy.getPositionName()} ${vacancy.getSalary()} руб/мес</a></li>
```
сделай vacancy/{id} path вместо get param
ид сразу после слэша

Taler777 committed:
bag fixed

Taler777 committed:
change == to equals

Taler777 committed:
status check to null and empty string fixed

yakovlevlexandrthirdparty committed:
js

baimurzin commented on src/main/resources/templates/career.ftl:
```diff
@@ -3,11 +3,10 @@
     <meta charset="UTF-8">
     <title>Chat</title>
-    <link href="/css/main.css" rel="stylesheet">
+    <link href="/css/main.css" rel="stylesheet"
```
закрыть тэг

baimurzin committed:
Merge pull request #4 from baimurzin/student-cab-alexeys

Student cab alexeys

yakovlevlexandrthirdparty committed:
закрыл тег

yakovlevlexandrthirdparty committed:
Merge branch 'dev' into taskCareerPage

yakovlevlexandrthirdparty committed:
Merge branch 'dev' into taskCareerPage

yakovlevlexandrthirdparty committed:
.gitignore

yakovlevlexandrthirdparty committed:
Merge branch 'dev' of https://github.com/baimurzin/cabinet into dev

# Conflicts:
#	src/main/java/org/itlabel/cabinet/controller/StudentCabinetController.java
#	src/main/resources/templates/tasks.ftl

yakovlevlexandrthirdparty committed:
Merge branch 'dev' into taskCareerPage

# Conflicts:
#	src/main/java/org/itlabel/cabinet/controller/StudentCabinetController.java
#	src/main/resources/templates/tasks.ftl

yakovlevlexandrthirdparty committed:
18.09.18

yakovlevlexandrthirdparty committed:
поправил ссылки
